File: other/py_find_primes/src/find_primes.py
'''
Created on Dec 14, 2011

@author: pfl
'''

import math

def brute(nMax):
    
    for i in range(2,nMax):
        prime = True
        for j in range(2,i):
            if i%j == 0:
                prime = False
                
        if prime :
            print("%i"%i)
    return

def optimised_1(nMax):
    
    print("2")
    for i in range(3,nMax,2):
        prime = True
        for j in range(2,int( math.sqrt(i) )):
            if i%j == 0:
                prime = False
                
        if prime :
            print("%i"%i)
    return

optimised_1(100000)

File: other/py_find_primes/c_src/find_primes.c
/*
 ============================================================================
 Name        : find_primes.c
 Author      : pfl
 Version     :
 Copyright   : Your copyright notice
 Description : Hello World in C, Ansi-style
 ============================================================================
 */

#include <stdio.h>
#include <stdlib.h>
#include <math.h>

void brute(int nMax)
{
	int i,j;

		for( i = 3; i<=nMax; i++)
		{
			for( j=3; j<=i; j++)
			{
				if ( i%j == 0 )
					break;
			}
			if( i == j)
				printf( "%i\n",i);
		}
}

void optimised_1(int nMax)
{
	int i,j;
	char thisIsAPrime;

			for( i = 3; i<nMax; i=i+2)
			{
				thisIsAPrime = 1;
				for( j=3; j<=sqrt(i); j++)
				{
					if ( i%j == 0 )
					{
						thisIsAPrime = 0;
						break;
					}

				}
				if( thisIsAPrime )
					printf( "%i\n",i);
			}
}
